### ISSUE TITLE
Use `pacta.data.validation` functions to validate pacta data

### ISSUE THREAD

**Issue by jdhoffa** [RMI-PACTA/workflow.data.preparation#18]
Supersedes https://github.com/RMI-PACTA/pacta.data.preparation/issues/133

**Comment by cjyetman** [comment]
note, I'm currently leaning towards integrating validation checks for `workflow.data.preparation` *inputs* in `pacta.data.preparation`, while using `pacta.data.validation` validation checks for `workflow.data.preparation` *outputs*

**Comment by jdhoffa** [comment]
I definitely agree that `pacta.data.preparation` should adequately validate any input `data` and `args` passed to any of it's exported functions. In that sense, it validates only _inputs_

Then, separately, `pacta.portfolio.allocate` should probably ALSO have functions that adequately validate any input `data` or `args`. 

I wonder if `pacta.data.validation` as its own separate package is actually necessary? 

NIT:
I am imagining a design broadly like this:
Consider some data-pipeline type `pacta.*` package. You may have a function called `prepare_*_data(data, arg1, arg2)`
``` r
@export
calculate_some_data <- function(data, arg1, arg2) {
   check_some_data_args(arg1, arg2)
   data %>%
      do_something(arg1, arg2)
}

@export
check_some_data_input <- function(arg1, arg2) {
   check(arg1)
   check(arg2)
}
```

then, if we WANT to pre-run all of the checks in a workflow, that can be easily facilitated by whoever manages the workflow by:
``` r
# I don't want to wait until the function is called because this thing takes 6 hours to run, so I am going to double check all my args are valid now
arg1 <- config$arg1
arg2 <- config$arg2

pacta.package::check_some_data_args(arg1, arg2)
# do a bunch of stuff
pacta.package::calculate_some_data(data, arg1, arg2)
```

So in that paradigm:
`pacta.data.preparation` would handle all of it's own checks and expectations, and export them to facilitate pre-checking in a `workflow`.
`pacta.portfolio.allocate` would do the same. 

This ensures that each package can adequately and independently handle all of it's own expectations.
But by exporting those checks, it also facilitates that a `workflow.*` developer may decide "Hey, this takes 6 hours, and I don't want to wait that long before checking stuff, so I'll just call the checks early"

What do you think?

**Comment by cjyetman** [comment]
Yes, I'm currently definitely in favor of re-combining packages that we once split apart.

**Comment by jdhoffa** [comment]
for the record `pacta.data.validation` was your brain-child, I always had the opinion that input validation should occur in the exported function that manipulates the data 😂

**Comment by cjyetman** [comment]
> I wonder if `pacta.data.validation` as its own separate package is actually necessary?

In the spirit of DRY, it's likely that many of the data objects that pacta.data.validation checks would be used by multiple functions/packages/repos/workflows, that was my reasoning for a separate package.

**Comment by jdhoffa** [comment]
Yeah fair enough, that does make sense!
